src/envoy/tcp/metadata_exchange/metadata_exchange.cc

```diff
@@ -295,6 +299,15 @@ std::string MetadataExchangeFilter::getMetadataId() {
   return local_info_.node().id();
 }
 
+void MetadataExchangeFilter::setMetadataNotFoundFilterState() {
+  const std::string key =
+      config_->filter_direction_ == FilterDirection::Downstream
+          ? DownstreamMetadataIdKey
```

Comment:
Why not just empty? Does empty have some special meaning? It is quite confusing to me that `key == value` represents not found.

Reply:
Because getValue can return false for empty value I think as it check hasValue which can return true for empty value I think..
We can say something like NotFound if it makes easier?

Reply:
Changed it to envoy.wasm.metadata_exchange.not_found

Reply:
@kyessenov If metadata is not set, `getValue({"filter_state", peer_metadata_id_key}` will return false, no? If this is the case then we would not need to introduce this new hard coded value to indicate metadata not found.

Reply:
Just discussed offline. We need an indicator that `peer is not capable of sending metadata`, which is what this why this filter metadata is set as what it is now. The question left is can we just use an empty string, instead of a hard coded `not found` string.

Reply:
using "empty" state in filterstate to indicate `peer is not capable of sending metadata` but setting peer_id to not found in node_info_cache, so that when calling report we can distinguish between metadata not found from peer not capable of sending metadata case.

Reply:
@bianpengyuan : I think best would be to set this to "unknown" like the way we set for other properties when we don't know them as compared to setting it to empty.. what do you think?

Reply:
I am fine making it explicit, and since we are making it explicit, let's make the condition more clear, so instead of unknown, how about `envoy.wasm.metadata_exchange.peer_incapable`?

Reply:
how about `envoy.wasm.metadata_exchange.peer_unknown` we are not sure if it's capable or not.. we just dont know about it?